test/testdata/infer/massign_return_rhs.rb

```diff
@@ -0,0 +1,11 @@
+# typed: true
+
+extend T::Sig
+
+sig {returns(T.nilable(T::Array[Integer]))}
+def foo
+  nil
+end
+
+T.reveal_type((a, b = foo)) # error: Revealed type: `T.nilable(T::Array[Integer])`
+T.reveal_type((c, *, d = 100)) # error: Revealed type: `Integer(100)`
```

Comment:
Can you make an `ast.exp` snapshot for this test? I'd like to see what the desugared output looks like.

Reply:
Can you also specifically add the example listed in #981 with unreachable code?